src/handler/sign_tx.c

```diff
@@ -474,13 +515,23 @@ bool receive_data(buffer_t *cdata, uint8_t chunk) {
         }
 
         init_sign_tx_ctx();
+        uint16_t error_code = 0;
         // read change output and sighash
-        read_change_info(cdata);
+        error_code = read_change_info(cdata);
+        if (error_code != 0) {
+            explicit_bzero(&G_context, sizeof(G_context));
+            THROW(error_code);
```

Comment:
Not cleaning the state was making the next call to the Ledger fail due to bad state.
It was very unlikely to throw an error here since the transaction would need to be extremely malformed.
But since we added stricter rules for the change list we may reach this error more often, hence the need to make this context cleanup.